### PR TITLE
python3Packages.mplfinance: init at 0.12.7a7

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

This package was a feature of Matplotlib that was removed in version 2.2.0.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/mplfinance/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  checkInputs = [ pytest tox ];
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-pATiprTELt8GrEkeKjILPfpdNDVoex5t+Mc+6Gg7cPY=";
+  };
+
+  meta = with lib; {
+    description =
+      "Matplotlib utilities for the visualization, and visual analysis, of financial data";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mplfinance/default.nix:17]: Personally not a fan of this line break.

File: pkgs/development/python-modules/mplfinance/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildPythonPackage, fetchPypi, matplotlib, pandas, pytest, tox }:
+
+buildPythonPackage rec {
+  pname = "mplfinance";
+  version = "0.12.7a7";
+
+  propagatedBuildInputs = [ matplotlib pandas ];
+  checkInputs = [ pytest tox ];
+
+  src = fetchPypi {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mplfinance/default.nix:10]: Usually src comes before inputs.

File: pkgs/development/python-modules/mplfinance/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildPythonPackage, fetchPypi, matplotlib, pandas, pytest, tox }:
+
+buildPythonPackage rec {
+  pname = "mplfinance";
+  version = "0.12.7a7";
+
+  propagatedBuildInputs = [ matplotlib pandas ];
+  checkInputs = [ pytest tox ];
```
[the remark this answers is not in the document]
Comment (ehmry) [reply pkgs/development/python-modules/mplfinance/default.nix:8]: No real tests here it seems.

File: pkgs/development/python-modules/mplfinance/default.nix (hunk 1 above)
Comment (ehmry) [reply pkgs/development/python-modules/mplfinance/default.nix:17]: `nixfmt` does it to reduce the line length.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/mplfinance/default.nix:17]: thats the reason why I switched to nixpkgs-fmt which does not do such line breaks.

File: pkgs/development/python-modules/mplfinance/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  meta = with lib; {
+    description =
+      "Matplotlib utilities for the visualization, and visual analysis, of financial data";
+    homepage = "http://github.com/matplotlib/mplfinance";
+    license = [
+      (licenses.free // {
+        url = "https://github.com/matplotlib/mplfinance/blob/master/LICENSE";
+      })
+    ];
+    mainainter = [ maintainers.ehmry ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mplfinance/default.nix:24]:
```suggestion
    mainainters = [ maintainers.ehmry ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
that why the pipeline fails

File: pkgs/development/python-modules/mplfinance/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/mplfinance/default.nix:19]: @ehmry I think you missed this one.

File: pkgs/development/python-modules/mplfinance/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-pATiprTELt8GrEkeKjILPfpdNDVoex5t+Mc+6Gg7cPY=";
+  };
+
+  propagatedBuildInputs = [ matplotlib pandas ];
+
+  meta = with lib; {
+    description =
+      "Matplotlib utilities for the visualization, and visual analysis, of financial data";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mplfinance/default.nix:16]:
```suggestion
    description = "Matplotlib utilities for the visualization, and visual analysis, of financial data";
```

File: pkgs/development/python-modules/mplfinance/default.nix [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  pname = "mplfinance";
+  version = "0.12.7a7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-pATiprTELt8GrEkeKjILPfpdNDVoex5t+Mc+6Gg7cPY=";
+  };
+
+  propagatedBuildInputs = [ matplotlib pandas ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mplfinance/default.nix:13]:
```suggestion

  # tests are only included on GitHub where this version misses a tag
  # and half of them fail
  doCheck = false;

  pythonImportsCheck = [ "mplfinance" ];

```

**Comment by SuperSandro2000** [comment]
I took the bravery and fixed all the issues for you.

**Pull request merged**